src/test/rpc/Simulate_test.cpp

```diff
@@ -0,0 +1,974 @@
+
+            // test without autofill
+            testTx(env, tx, validateOutput);
+
+            // TODO: check that the ledger wasn't affected
+        }
+    }
+
+    void
+    testTransactionSigningFailure()
+    {
+        testcase("Transaction with a key-related failure");
+
+        using namespace jtx;
+        Env env(*this);
+        static auto const newDomain = "123ABC";
+        Account const alice{"alice"};
+        env(regkey(env.master, alice));
+        env(fset(env.master, asfDisableMaster), sig(env.master));
+        env.close();
+
+        {
+            std::function<void(Json::Value const&, Json::Value const&)> const&
+                testSimulation =
+                    [&](Json::Value const& resp, Json::Value const& tx) {
+                        auto result = resp[jss::result];
+                        checkBasicReturnValidity(
+                            result, tx, 3, env.current()->fees().base);
+
+                        BEAST_EXPECT(
+                            result[jss::engine_result] == "tefMASTER_DISABLED");
+                        BEAST_EXPECT(result[jss::engine_result_code] == -188);
+                        BEAST_EXPECT(
+                            result[jss::engine_result_message] ==
+                            "Master key is disabled.");
+
+                        BEAST_EXPECT(
+                            !result.isMember(jss::meta) &&
+                            !result.isMember(jss::meta_blob));
+                    };
+
+            Json::Value tx;
+
+            tx[jss::Account] = env.master.human();
+            tx[jss::TransactionType] = jss::AccountSet;
+            tx[sfDomain] = newDomain;
+
+            // test with autofill
+            testTx(env, tx, testSimulation, 3);
+
+            tx[sfSigningPubKey] = "";
+            tx[sfTxnSignature] = "";
+            tx[sfSequence] = env.seq(env.master);
+            tx[sfFee] = env.current()->fees().base.jsonClipped().asString();
+
+            // test without autofill
+            testTx(env, tx, testSimulation, 3);
+
+            // TODO: check that the ledger wasn't affected
+        }
+    }
+
+    void
+    testMultisignedBadPubKey()
+    {
+        testcase("Multi-signed transaction with a bad public key");
+
+        using namespace jtx;
+        Env env(*this);
+        static auto const newDomain = "123ABC";
+        Account const alice("alice");
+        Account const becky("becky");
+        Account const carol("carol");
+        Account const dylan("dylan");
+        env.fund(XRP(10000), alice);
+        env.close();
+
+        // set up valid multisign
+        env(signers(alice, 1, {{becky, 1}, {carol, 1}}));
+
+        {
+            auto validateOutput = [&](Json::Value const& resp,
+                                      Json::Value const& tx) {
+                auto result = resp[jss::result];
+                checkBasicReturnValidity(
+                    result, tx, 5, env.current()->fees().base * 2);
```

Comment:
Please change 5 to seq()

Reply:
fixed in 0bbf589f53a13069448d709faa5cea88b6ee547f